### PR TITLE
[libc] Introduce generic implementation of bcopy, memcpy, memmove and strlen

### PR DESCRIPTION
**Pull request by MichalBlk**
Some architectures don't supply specialized implementation of aforementioned functions in NetBSD's libc.
These architectures rely on generic implementation written in C. RISC-V is among them. With generic implementation in `gen`
and presented changes in `common.mk`, one can simply make a symlink named `$(ARCH)` to `gen`.

### CODE REVIEW

File: lib/libc/string/gen/strlen.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+#include <sys/cdefs.h>
+
+#if !defined(_KERNEL) && !defined(_STANDALONE)
+#include <assert.h>
+#include <string.h>
+#else
+#include <sys/libkern.h>
+#endif
```
Comment (pj1031999) [inline lib/libc/string/gen/strlen.c:39]: Do we need to include anything for that simple function?

File: lib/libc/string/gen/memcpy.c [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/*	$NetBSD: memcpy.c,v 1.2 2013/12/02 21:21:33 joerg Exp $	*/
+
+#define MEMCOPY
+#include "bcopy.c"
+
+#if defined(__ARM_EABI__)
+__strong_alias(__aeabi_memcpy, memcpy)
+#endif
```
Comment (pj1031999) [inline lib/libc/string/gen/memcpy.c:8]: Is it used?

File: lib/libc/string/gen/memmove.c [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/*	$NetBSD: memmove.c,v 1.2 2013/12/02 21:21:33 joerg Exp $	*/
+
+#define MEMMOVE
+#include "bcopy.c"
+
+#if defined(__ARM_EABI__)
+__strong_alias(__aeabi_memmove, memmove)
+  __strong_alias(__aeabi_memmove4, memmove)
+    __strong_alias(__aeabi_memmove8, memmove)
+#endif
```
Comment (pj1031999) [inline lib/libc/string/gen/memmove.c:10]: ditto

**Review by pj1031999 — COMMENTED** [review]
I would opt for removing unused implementation from bcopy.c.

File: lib/libc/string/gen/strlen.c (hunk 1 above)
Comment (MichalBlk) [reply lib/libc/string/gen/strlen.c]: You're right. Just `stddef.h` will do.

File: lib/libc/string/gen/memcpy.c (hunk 2 above)
Comment (MichalBlk) [reply lib/libc/string/gen/memcpy.c]: It isn't. We won't use ARM EABI, so the code has been removed.

File: lib/libc/string/gen/memmove.c (hunk 3 above)
Comment (MichalBlk) [reply lib/libc/string/gen/memmove.c]: Removed.

**Comment by MichalBlk** [comment]
The code has been cleaned. FTTB, I left the size optimized implementations and I will consult @cahirwpz if I should remove them.

File: build/common.mk [hunk 4]
```diff
@@ -68,7 +68,7 @@ include $(TOPDIR)/build/tools.mk
 
 SRCDIR ?= .
 
-vpath %.c $(SRCDIR)
+vpath %.c $(SRCDIR) $(SRCDIR)/$(ARCH)
```
Comment (cahirwpz) [inline build/common.mk:71]: What is the purpose of this change? Is it really needed?

**Review by cahirwpz — APPROVED** [review]
